tools/downloader/common.py

```diff
@@ -300,30 +295,57 @@ def file_precision(file):
             return cls(name, subdirectory, files, postprocessing, mo_args, framework,
                 description, license_url, precisions, task_type)
 
-def load_topologies(config):
-    with config.open() as config_file:
-        try:
-            topologies = []
-            topology_names = set()
+def load_topologies(args):
+    topologies = []
+    topology_names = set()
+
+    def add_topology(topology):
+        topologies.append(topology)
+
+        if topologies[-1].name in topology_names:
+            raise DeserializationError(
+                'Duplicate topology name "{}"'.format(topologies[-1].name))
```

Comment:
According to that it will not be possible to add the model with the same name to different paths. Is that intentionally expected? If yes, what is the path to store info for different flavors of the same model, let's say densenet-zzz / CF and TF. Should it be stored in the same folder/config than or sub-folders within densenet-zzz?
why not to use full path as a key?

Reply:
>  Is that intentionally expected?

Yes.

> If yes, what is the path to store info for different flavors of the same model, let's say densenet-zzz / CF and TF. Should it be stored in the same folder/config than or sub-folders within densenet-zzz?

Nothing changes in that regard from how it already works. Different flavors of DenseNet are considered different models and have different names.

> why not to use full path as a key?

This is possible; however, it would change the names of all existing models. I don't think this change is beneficial enough to be worthwhile.